### ISSUE TITLE
pagesNav helper showing all pages even when auth filters down the collection to one element

### ISSUE THREAD

**Issue by douglascamata** [alethes/meteor-pages#108]
Code sample to reproduce it:

```javascript
MessagesPages = new Meteor.Pagination(Messages, {
  perPage: 3, // small number just to don't need to create that much items in the collection
  auth: function () {
    return [{text: 'foo'}, {}] // any filter to narrow down the result to 1 element
  }
});```

**Comment by alethes** [comment]
Thanks for reporting. I checked, and it seems the issue no longer occurs in the most recent version.

**Comment by Shelagh-Lewins** [comment]
I'm seeing this bug in version 1.8.4. My code is like this:

    this.MyPatterns = new Meteor.Pagination(Patterns, {
    auth: function(skip, sub){
        return Patterns.find({created_by: sub.userId}); // nav shows for all results not just those created by user
      }
    });

The first page shows the correctly filtered results which are less than 1 page. But the Navigation is shown and when I click any other page than 1, I see an empty page.

Note that I do see the correct behaviour if instead of directly returning a cursor, auth returns parameters:

    auth: function(skip, sub){
      var _filters = {created_by: sub.userId};
      var _options = {sort: {name: 1}};
      return [_filters, _options];
        return Patterns.find({created_by: sub.userId}); // works correctly
      }

It's a generally awesome plugin, thank you.

**Comment by oskareke** [comment]
I'm having the same problem - did you find a solution by any chance?

**Comment by Shelagh-Lewins** [comment]
My workaround is to have auth return filters not a cursor, as shown in my second code example. (I just fixed an error in that code example.)

It seems to be a separate issue that the nav is shown before the results appear, making it appear briefly at the top of the screen. You can put the nav above the results, but this is not normal practice and isn't where users will expect it to be. It would be great if there was some helper for the template equivalent to "subscriptionsReady" that would allow me to hide the nav until the results were available.

**Comment by oskareke** [comment]
Perfect, thank you!

Agree that a subscriptionsReady callback would be great.

**Comment by Shelagh-Lewins** [comment]
It seems that with the workaround, you cannot then add client side filters or sort, which is a shame because it's such a nice feature to allow the user to filter or sort results. At least, I couldn't get it to work when auth is set in this way.

**Comment by oskareke** [comment]
@Shelagh-Lewins If you include the filters in the auth-function you can combine the two. 
Solution by @fletchtj in https://github.com/alethes/meteor-pages/issues/131 :

auth: function (skip, sub) {
    if (!sub.userId) { return false; }
    var _orgId = App.users.getOrgId(sub.userId)
        , userSettings = UserSearchResults.userSettings[sub._session.id] || {}
        , uFilters = userSettings.filters || this.filters
        , uFields = userSettings.fields || this.fields
        , uSort = userSettings.sort || this.sort
        , uPerPage = userSettings.perPage || this.perPage
        , _filters = _.extend({}, uFilters, { orgId: _orgId })
        , _options = { fields: uFields, sort: uSort, limit: uPerPage, skip: skip };
    return [ _filters, _options ];
}

**Comment by Shelagh-Lewins** [comment]
Thank you so much oskareke! I got that to work. I want to show only documents created by the user, and then allow the user to filter further on the client side. My JavaScript auth is:

    auth: function(skip, sub){
        var userFilters = this.userSettings[sub._session.id].filters;
        var _filters = _.extend({created_by: sub.userId}, userFilters);
        return [_filters, _options];
      }

I couldn't see anything in the readme about userSettings, so I'm very grateful for your help, I would not have known about this otherwise.

**Comment by Shelagh-Lewins** [comment]
Ah, I have just found I missed something important in my answer above; if you do not pass in 'skip' to the options then you will not see anything on pages after 1.

    auth: function(skip, sub){
      if (typeof this.userSettings[sub._session.id] !== "undefined")
        var userFilters = this.userSettings[sub._session.id].filters;
      else
        var userFilters = {};

      var _options = {skip: skip};

      var _filters = _.extend({created_by: sub.userId}, userFilters);
      return [_filters, _options];
    }

Basically the auth replaces ALL your setup options - anything you want to use must be passed into the auth. But the skip is dynamic and is automatically passed in from the user's selection in the nav.
